include/open62541/server.h

```diff
@@ -670,7 +670,47 @@ typedef struct {
     void (*destructor)(UA_Server *server,
                        const UA_NodeId *sessionId, void *sessionContext,
                        const UA_NodeId *nodeId, void *nodeContext);
-} UA_GlobalNodeLifecycle;
+
+    /* Can be NULL. Called during recursive node instantiation. While mandatory
+     * child nodes are automatically created if not already present, optional child
+     * nodes are not. This callback can be used to define whether an optional child
+     * node should be created.
+     *
+     * @param server The server executing the callback
+     * @param sessionId The identifier of the session
+     * @param sessionContext Additional data attached to the session in the
+     *        access control layer
+     * @param sourceNodeId Source node from the type definition. If the new node
+     *        shall be created, it will be a copy of this node.
+     * @param targetParentNodeId Parent of the potential new child node
+     * @param referenceTypeId Identifies the reference type which that the parent
+     *        node has to the new node. */
+    UA_Boolean (*onCreateOptionalChild)(UA_Server *server,
+                                        const UA_NodeId *sessionId,
+                                        void *sessionContext,
+                                        const UA_NodeId *sourceNodeId,
+                                        const UA_NodeId *targetParentNodeId,
+                                        const UA_NodeId *referenceTypeId);
+
+    /* Can be NULL. Called when a node is to be copied during recursive
+     * node instantiation. Allows definition of the NodeId for the new node.
+     * If this callback is not used, random numeric NodeIds will be generated.
```

Comment:
We can have the callback **and** allow random NodeId generation.
Rephrase as follows:
If the callback is set to NULL or the resulting NodeId is UA_NODEID_NULL, then a random NodeId will be generated.